### PR TITLE
PDOStatement::fetchAll() has incorrect type

### PR DESCRIPTION
**Pull request by hc-jworman**
Closes https://github.com/vimeo/psalm/issues/7784

### CODE REVIEW

**Comment by orklah** [comment]
Thanks!

Tests will show that you should also change that in Callmap.php (that is the current state of the signatures). Can you do the change?

**Comment by hc-jworman** [comment]
Done

**Comment by orklah** [comment]
Thanks! Congrats on your first PR!

File: dictionaries/CallMap.php [hunk 1]
```diff
@@ -9911,7 +9911,7 @@
 'PDOStatement::errorInfo' => ['array{0: ?string, 1: ?int, 2: ?string, 3?: mixed, 4?: mixed}'],
 'PDOStatement::execute' => ['bool', 'params='=>'?array'],
 'PDOStatement::fetch' => ['mixed', 'mode='=>'int', 'cursorOrientation='=>'int', 'cursorOffset='=>'int'],
-'PDOStatement::fetchAll' => ['array|false', 'mode='=>'int', '...args='=>'mixed'],
+'PDOStatement::fetchAll' => ['array', 'mode='=>'int', '...args='=>'mixed'],
```
[the remark this answers is not in the document]
Comment (hc-jworman) [reply dictionaries/CallMap.php:9914]: The documentation states "CallMap.php contains current signatures (for PHP 8.1 at the time of writing)".

[the remark this answers is not in the document]
Comment (staabm) [reply dictionaries/CallMap.php:9914]: Hm ok. In phpstan this is different. Sorry for the noise then

**Pull request merged**